### ISSUE TITLE
Replace deprecated `eslint` style/formatting directives with ESlint Stylistic

### ISSUE THREAD

**Issue by miketheman** [pypi/warehouse#15274]
Our current setup runs `eslint` on any javascript files, finding linter errors and addressing stylistic concerns.

eslint has decided to deprecate formatting rules - see https://github.com/eslint/eslint/issues/17522 for more context.

The existing formatters have been extracted, and will be updated outside of eslint core.
See more details at https://eslint.style/ and specifically https://eslint.style/packages/js (we don't have any typescript or jsx in this project, so we don't need the meta-plugin)

To accomplish this, review stylistic, learn about the migration path, and apply necessary changes - `package.json` is a good place to start.
We probably need to update some javascript packages - we don't have dependabot do that for us regularly yet.

The end goal is to have `npm run lint` successfully and minimize the resulting code churn, as we have some of our custom styles set in `package.json`

---

**Good First Issue**: This issue is good for first time contributors. If you've already contributed to Warehouse, work on [another issue without this label](https://github.com/pypi/warehouse/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22good+first+issue%22) instead. If there is not a corresponding pull request for this issue, it is up for grabs. For directions for getting set up, see our [Getting Started Guide](https://warehouse.pypa.io/development/getting-started/).

If you are working on this issue and have questions, feel free to ask them here, in the [`#pypa-dev` chat channel on Libera](https://web.libera.chat/#pypa-dev), the [PyPA Discord](https://discord.gg/pypa) or on the [Discourse](https://discuss.python.org/c/packaging/14).

**Comment by jzohrab** [comment]
Hey there, wondering if I can make a small dent in this -- at least my questions may help someone else finish this if I can't.

From the [getting started](https://warehouse.pypa.io/development/getting-started.html) there was no obvious way to do `npm run lint` in docker, so I'm assuming that `make lint` is the right thing to do.

Currently the lint runs successfully:

<detail>
<summary>Lint output</summary>

```
$ make lint
# Build our base container for this project.
... snip ...
...
touch .state/docker-build-base
docker compose run --rm base bin/lint
WARN[0000] Found orphan containers ([warehouse-vault-1]) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up. 
[+] Creating 1/0
 ✔ Volume "warehouse_rstuf-metadata"  Created                                                                                     0.0s 
bin/lint: line 7: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
+ find . -name '*.py' -exec python -m pyupgrade --py311-plus '{}' +
+ python -m flake8 .
+ python -m black --check --diff gunicorn-prod.conf.py gunicorn-uploads.conf.py warehouse/ tests/
All done! ✨ 🍰 ✨
703 files would be left unchanged.
+ python -m isort --check gunicorn-prod.conf.py gunicorn-uploads.conf.py warehouse/ tests/
Skipped 2 files
+ python -m doc8 --allow-long-titles README.rst CONTRIBUTING.rst docs/ --ignore-path 'docs/**/_build/'
Scanning...
Validating...
========
Total files scanned = 28
Total files ignored = 0
Total accumulated errors = 0
Detailed error counts:
    - doc8.checks.CheckCarriageReturn = 0
    - doc8.checks.CheckIndentationNoTab = 0
    - doc8.checks.CheckMaxLineLength = 0
    - doc8.checks.CheckNewlineEndOfFile = 0
    - doc8.checks.CheckTrailingWhitespace = 0
    - doc8.checks.CheckValidity = 0
+ python -m curlylint ./warehouse/templates ./docs/blog
All done! ✨ 🍰 ✨

+ python -m mypy -p warehouse
Success: no issues found in 368 source files
+ ./bin/flushes
/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
docker compose run --rm static bin/static_lint
WARN[0000] Found orphan containers ([warehouse-vault-1]) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up. 
+ export LC_ALL=en_US.UTF-8
+ LC_ALL=en_US.UTF-8
bin/static_lint: line 8: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ npm run lint

> lint
> eslint 'warehouse/static/js/**' 'tests/frontend/**' --ignore-pattern 'warehouse/static/js/vendor/**'

npm notice 
npm notice New minor version of npm available! 10.1.0 -> 10.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.0
npm notice Run npm install -g npm@10.5.0 to update!
npm notice 
+ npm run stylelint

> stylelint
> stylelint '**/*.scss' --cache

MacBook-Pro:warehouse jeff$ 

```

</detail>

re "learn about the migration path", you mean https://eslint.style/guide/migration ?

**Comment by miketheman** [comment]
Resolved by #15689 , thanks @jzohrab !
